### PR TITLE
Feature/6 automatic config - workflows

### PR DESCRIPTION
**Pull request by adrrf**
This pull request includes updates to the `.github/CODEOWNERS` file and adds a new GitHub Actions workflow for syncing OpenAPI specs. Additionally, a new script `sync-openapi.sh` is introduced to fetch OpenAPI specifications for various services.

Key changes:

* Updates to code owners:
  * [`.github/CODEOWNERS`](diffhunk://#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7L1-R1): Changed the code owner from `@alvarobernal2412` to `@AntonioRodriguezRuiz`.

* New GitHub Actions workflow:
  * [`.github/workflows/sync-openapi.yml`](diffhunk://#diff-73e05868d04fb487b337e310e72b8fd579a005129e413d0959a6388110c76e8dR1-R23): Added a new workflow to sync OpenAPI specs, which includes steps for checking out the repository, running the `sync-openapi.sh` script, and committing and pushing changes.

* New script for syncing OpenAPI specs:
  * [`sync-openapi.sh`](diffhunk://#diff-8322cb01f0d17bed7c8c68c5bb682b7c2ec2c5fb8d7b983e628e4bc32c9f618fR1-R28): Created a script to fetch OpenAPI specifications for a list of services from a specified base URL and save them in the `docs` directory. The script also logs the date of the last update.

### CODE REVIEW

File: .github/workflows/sync-openapi.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        run: |
+          mkdir -p docs
+          chmod +x ./sync-openapi.sh
+          ./sync-openapi.sh
+      - name: Commit and push changes 📤
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+          git add docs
+          git diff --quiet && git diff --staged --quiet || (gir commit -m "chore: sync openapi specs🔄" && git push)
```
Comment (alvarobernal2412) [inline .github/workflows/sync-openapi.yml]: I think `gir commit` is a typo, isn't it?

File: .github/workflows/sync-openapi.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  update-specs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Sync OpenAPI Specs 🔄
+        run: |
+          mkdir -p docs
+          chmod +x ./sync-openapi.sh
```
Comment (alvarobernal2412) [inline .github/workflows/sync-openapi.yml]: If you give permissions to the file and push it to the remote, you don't need to do this always in the workflow.

File: .github/workflows/sync-openapi.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+on:
+  repository_dispatch:
+    types: [spec-sync]
+  workflow_dispatch:
+
+jobs:
+  update-specs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (alvarobernal2412) [inline .github/workflows/sync-openapi.yml]: This checkout version is old, i urge you to update it to a new one.

**Review by alvarobernal2412 — CHANGES_REQUESTED** [review]
It looks nice but it would be perfect with some fixes!

File: .github/workflows/sync-openapi.yml (hunk 1 above)
Comment (adrrf) [reply .github/workflows/sync-openapi.yml]: yup

**Review by alvarobernal2412 — APPROVED** [review]
LGTM!😈

**Review by TunixR — APPROVED** [review]
LGTM

**Pull request merged**